src/apps/bridge/sensor_drivers/softSensor.cpp

```diff
@@ -0,0 +1,138 @@
+        uint32_t reading_time_millis = soft_data.header.reading_time_utc_ms % 1000U;
+        uint64_t sensor_reading_time_sec = soft_data.header.sensor_reading_time_ms / 1000U;
+        uint32_t sensor_reading_time_millis = soft_data.header.sensor_reading_time_ms % 1000U;
+
+        size_t log_buflen =
+            snprintf(log_buf, SENSOR_LOG_BUF_SIZE,
+                     "%" PRIx64 ","   // Node Id
+                     "%" PRIu64 ","   // reading_uptime_millis
+                     "%" PRIu64 "."   // reading_time_utc_ms seconds part
+                     "%03" PRIu32 "," // reading_time_utc_ms millis part
+                     "%" PRIu64 "."   // sensor_reading_time_ms seconds part
+                     "%03" PRIu32 "," // sensor_reading_time_ms millis part
+                     "%.3f\n",        // temp_deg_c
+                     node_id, soft_data.header.reading_uptime_millis, reading_time_sec,
+                      reading_time_millis, sensor_reading_time_sec, sensor_reading_time_millis,
+                      soft_data.temperature_deg_c);
+        if (log_buflen > 0) {
+          // TODO - keep as individual log or switch to the bm_sensor common log
+          BRIDGE_SENSOR_LOG_PRINTN(SOFT_IND, log_buf, log_buflen);
+        } else {
+          printf("ERROR: Failed to print soft individual log\n");
+        }
+        vPortFree(log_buf);
+      }
+      xSemaphoreGive(soft->_mutex);
+    } else {
+      printf("Failed to get the subbed Soft mutex after getting a new reading\n");
+    }
+  }
+}
+
+void SoftSensor::aggregate(void) {
+  char *log_buf = static_cast<char *>(pvPortMalloc(SENSOR_LOG_BUF_SIZE));
+  configASSERT(log_buf);
+  if (xSemaphoreTake(_mutex, portMAX_DELAY)) {
+    size_t log_buflen = 0;
+    soft_aggregations_t soft_aggs = {.temp_mean_deg_c = NAN, .reading_count = 0};
+    // Check to make sure we have enough sensor readings for a valid aggregation.
+    // If not send NaNs for all the values.
+    if (temp_deg_c.getNumSamples() >= MIN_READINGS_FOR_AGGREGATION) {
+      soft_aggs.temp_mean_deg_c = temp_deg_c.getMean();
+      soft_aggs.reading_count = reading_count;
+      if (soft_aggs.temp_mean_deg_c < TEMP_SAMPLE_MEMBER_MIN ||soft_aggs.temp_mean_deg_c > TEMP_SAMPLE_MEMBER_MAX) {
+        soft_aggs.temp_mean_deg_c = NAN;
+      }
+    }
+    static constexpr uint8_t TIME_STR_BUFSIZE = 50;
+    char time_str[TIME_STR_BUFSIZE];
+    if (!logRtcGetTimeStr(time_str, TIME_STR_BUFSIZE, true)) {
+      printf("Failed to get RTC time string for Soft aggregation\n");
+      snprintf(time_str, TIME_STR_BUFSIZE, "0");
+    }
+    log_buflen = snprintf(log_buf, SENSOR_LOG_BUF_SIZE,
+                          "%s,"          // timstamp(ticks/UTC)
+                          "%" PRIx64 "," // Node Id
+                          "%" PRIu32 "," // reading_count
+                          "%.3f\n",      // temp_mean_deg_c
+                          time_str, node_id, soft_aggs.reading_count, soft_aggs.temp_mean_deg_c);
```

Comment:
log the reading count for now. Will decide on removing reading count when the shared log stuff is done.